src/js/utils/date-utils.js

```diff
@@ -0,0 +1,258 @@
+/**
+ * Create a formatted, locale-aware string representation of a datetime.
+ *
+ * This module contains general purpose functions for localizing and displaying
+ * date-times by timezone, locale and preferred language."
+ *
+ * Most of the heavy lifting is done by the {@link http://momentjs.com/ |'moments' js library}
+ * with the moment-with-locales and moment-timezones add-ons.
+ *
+ *
+ *
+ * Example:
+ *
+ *~~~ javascript
+ * var context = {
+ *    datetime: '2016-10-14 08:00:00+00:00',
+ *    timezone: 'Pacific/Kiritimati',
+ *    language: 'ar',
+ *    format: 'DEFAULT_LONG_DATE_FORMAT'
+ *    };
+ * return_string = DateUtils.localize(context);
+ *~~~
+ *
+ * @module DateUtils
+ */
+
+
+(function(define) {
+    'use strict';
+    define([
+        'jquery',
+        'moment',
+        'moment-timezone'
+    ], function(
+        $,
+        moment,
+        momentTZ
+        ) {
+        var DateUtils,
+            localize,
+            stringToMoment,
+            localizeTime,
+            getSysTimezone,
+            getDisplayTime,
+            getDisplayLang,
+            getDisplayFormat,
+            isValid;
+
+        var DATE_FORMATS = {
+            DEFAULT_SHORT_DATE_FORMAT: 'll', // example: Jan 01, 2016
+            DEFAULT_LONG_DATE_FORMAT: 'LLLL', // example: Friday, January 01, 2016
+            DEFAULT_TIME_FORMAT: 'LTS z', // example: Jan 01, 2016 15:30 UTC
+            DEFAULT_DATE_TIME_FORMAT: 'll HH[:]mm z' // example: Jan 01, 2016 15:30 UTC
+        };
+
+        var DEFAULT_LANG = 'en';
+        var DEFAULT_TIMEZONE = 'UTC';
+        /**
+         *
+         * Renders a localized date/time in a standard format.
+         *
+         * Please Note: the preferred method is via underscore in the mako
+         * templates, however this will also work.
```

Comment:
?? Why is the preferred method in the mako template?  I thought this was the new recommended method?